pennylane/qnode.py

```diff
@@ -350,14 +350,21 @@ def construct(self, args, kwargs=None):
 
         # quantum circuit function return validation
         if isinstance(res, pennylane.operation.Observable):
-            self.output_type = float
+            if res.return_type == "sample":
+                # Squeezing ensures that there is only one array of values returned
+                # when only a single-mode sample is requested
+                self.output_type = np.squeeze
```

Comment:
Good solution

Reply:
What do you think of renaming it from `self.output_type` to `self.convert_for_output` or something along those lines? `output_type` is clearly misleading

Reply:
Sounds good @johannesjmeyer, but perhaps we can bump that to a qnode refactor PR instead

Reply:
I'd just change it here, because it is used exclusively in this single method of the class (we don't even need the `self.` in front of it, actually)

Reply:
Okay, the fact that we don't need `self.` was a lie, but I changed the name to `self.output_conversion`